testcases/urls.json

```diff
@@ -0,0 +1,14 @@
+{
+    "specversion": "0.4-wip",
+
+    "id": "123",
+
+    // Source is a URI-reference, https://tools.ietf.org/html/rfc3986#appendix-A
+    "source": "",  // relative-ref -> relative-part -> path-empty
+    "type": "io.cloudevents.url-test",
+    
+    // URI-reference: https://tools.ietf.org/html/rfc3986#appendix-A
+    // Verify that % decoding is done properly. This is a 'path-noscheme'
+    // that percent-decodes to a '"//" authority path-abempty'.
+    "schemaurl": "%2f%2f%anonymous@example.com/a&b;?x'=%2f//#//",
```

Comment:
remove trailing `,`

Reply:
Do we need to say something in our spec about doing a percent-decoding?
I want to say 'no', but some kind of "heads-up" might be good, not from a `strcmp` perspective, but rather to remind people that per RFC3986 someone might URL-encoding them even if they're not meant to be deference-able. Perhaps something in the primer?

what do people think?

Reply:
This is a tricky case, because it _is_ a valid URL, but if you percent-decode it, it is not a valid URL, but instead a `hier-part`, which requires a `scheme ":"` prefix to be a valid `URI` production.

Clarified the comment.

Reply:
Hmm, it looks like I didn't look at this quite closely enough (but may perhaps need to clarify the HTTP spec as well):

https://github.com/cloudevents/spec/blob/master/http-transport-binding.md#3132-http-header-values

> Non-printable ASCII characters and non-ASCII characters MUST first be encoded according to UTF-8, and then each octet of the corresponding UTF-8 sequence MUST be percent-encoded to be represented as HTTP header characters, in compliance with RFC7230, sections 3, 3.2, 3.2.6. The rules for encoding of the percent character ('%') apply as defined in RFC 3986 Section 2.4.

However, the characters I encoded are not in the set (non-printable ASCII characters and non-ASCII characters), so it's not clear whether it is _allowed_ for them to be percent-encoded or not.

Thinking about this from a software-implementer point of view, it would be a nightmare to scan through each percent-encoded string and determine whether or not the character was from the (non-printable ASCII or non-ASCII) character set before deciding whether or not to process the escape sequence.

I'll clarify the HTTP transport bindings to make it clear that all HTTP Header Values should be decoded through **a single round** of percent-decoding, and update this PR when that lands.